api/src/main/java/graphql/nadel/instrumentation/parameters/NadelInstrumentationExecuteOperationParameters.java

```diff
@@ -49,13 +80,19 @@ public NadelInstrumentationExecuteOperationParameters(
      * @return a new parameters object with the new state
      */
     public NadelInstrumentationExecuteOperationParameters withNewState(InstrumentationState instrumentationState) {
-        return new NadelInstrumentationExecuteOperationParameters(normalizedQueryFromAst, document, graphQLSchema, variables, operationDefinition, instrumentationState, context);
+        return new NadelInstrumentationExecuteOperationParameters(normalizedQueryFromAst, normalizedOperation, document, graphQLSchema, variables, operationDefinition, instrumentationState, context);
     }
 
+    @Nullable
     public NormalizedQueryFromAst getNormalizedQueryFromAst() {
         return normalizedQueryFromAst;
     }
 
+    @Nullable
+    public ExecutableNormalizedOperation getNormalizedOperation() {
```

Comment:
Consumers will get an `ExecutableNormalizedOperation` when a query is executed on the next-gen engine. Consumers code in the `beginExecute` callback will need to handle both `NormalizedQueryFromAst` and `ExecutableNormalizedOperation`